### PR TITLE
Add `Choose a different Option` link to passkey, totp flows

### PR DESCRIPTION
**Pull request by ashanthamara**
image
image
image

### Related Issues
- https://github.com/wso2/product-is/issues/17962

### CODE REVIEW

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp [hunk 1]
```diff
@@ -32,6 +32,32 @@
...
+                isMultiAuthAvailable = false;
+            } else {
+                String authenticators = multiOptionURI.substring(authenticatorIndex + 15);
+                int authLastIndex = authenticators.indexOf("&") != -1 ? authenticators.indexOf("&") : authenticators.length();
+                authenticators = authenticators.substring(0, authLastIndex);
+                List<String> authList = new ArrayList<>(Arrays.asList(authenticators.split("%3B")));
+                if (authList.size() < 2) {
+                    isMultiAuthAvailable = false;
+                }
+                else if (authList.size() == 2 && authList.contains("backup-code-authenticator%3ALOCAL")) {
```
Comment (JayaShakthi97) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:52]:
```suggestion
                } else if (authList.size() == 2 && authList.contains("backup-code-authenticator%3ALOCAL")) {
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp [hunk 2]
```diff
@@ -150,14 +176,34 @@
...
-                            <div class="mt-3">
-                                <div class="buttons">
-                                    <button class="ui secondary fluid large button" type="button" onclick="cancelFlow()"
-                                    data-testid="login-page-fido-cancel-button">
-                                        <%=AuthenticationEndpointUtil.i18n(resourceBundle, "fido.cancel" )%>
-                                    </button>
+                            <%
+                                String multiOptionURI = request.getParameter("multiOptionURI");
+                                if (multiOptionURI != null && AuthenticationEndpointUtil.isValidURL(multiOptionURI) &&
+                                isMultiAuthAvailable(multiOptionURI)) {
```
Comment (JayaShakthi97) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:182]:
```suggestion
                                    isMultiAuthAvailable(multiOptionURI)) {
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-identifierfirst.jsp [hunk 3]
```diff
@@ -32,6 +32,32 @@
...
+                isMultiAuthAvailable = false;
+            } else {
+                String authenticators = multiOptionURI.substring(authenticatorIndex + 15);
+                int authLastIndex = authenticators.indexOf("&") != -1 ? authenticators.indexOf("&") : authenticators.length();
+                authenticators = authenticators.substring(0, authLastIndex);
+                List<String> authList = new ArrayList<>(Arrays.asList(authenticators.split("%3B")));
+                if (authList.size() < 2) {
+                    isMultiAuthAvailable = false;
+                }
+                else if (authList.size() == 2 && authList.contains("backup-code-authenticator%3ALOCAL")) {
```
Comment (JayaShakthi97) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-identifierfirst.jsp:52]:
```suggestion
                } else if (authList.size() == 2 && authList.contains("backup-code-authenticator%3ALOCAL")) {
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp (hunk 1 above)
Comment (ashanthamara) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp]: Fixed with https://github.com/wso2/identity-apps/pull/5261/commits/809419a821d00242ef890936b2101cbe317627cb

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp (hunk 2 above)
Comment (ashanthamara) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp]: Fixed with https://github.com/wso2/identity-apps/pull/5261/commits/809419a821d00242ef890936b2101cbe317627cb

**Review by JayaShakthi97 — APPROVED** [review]

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp [hunk 4]
```diff
@@ -32,6 +32,31 @@
 <%@include file="includes/localize.jsp" %>
 <%@include file="includes/init-url.jsp" %>
 
+<%!
+    private boolean isMultiAuthAvailable(String multiOptionURI) {
+        boolean isMultiAuthAvailable = true;
```
Comment (inthirakumaaran) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:37]: Add a new line after the method declaration, and check the other places as well.

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp [hunk 5]
```diff
@@ -32,6 +32,31 @@
...
+            isMultiAuthAvailable = false;
+        } else {
+            int authenticatorIndex = multiOptionURI.indexOf("authenticators=");
+            if (authenticatorIndex == -1) {
+                isMultiAuthAvailable = false;
+            } else {
+                String authenticators = multiOptionURI.substring(authenticatorIndex + 15);
+                int authLastIndex = authenticators.indexOf("&") != -1 ? authenticators.indexOf("&") : authenticators.length();
+                authenticators = authenticators.substring(0, authLastIndex);
+                List<String> authList = new ArrayList<>(Arrays.asList(authenticators.split("%3B")));
```
Comment (inthirakumaaran) [inline identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:48]: Removed unnecessary conversion of `authList` to an ArrayList:  
```
List<String> authList = Arrays.asList(authenticators.split("%3B"));
```

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp (hunk 4 above)
Comment (ashanthamara) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:37]: Fixed with https://github.com/wso2/identity-apps/pull/5833

File: identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp (hunk 5 above)
Comment (ashanthamara) [reply identity-apps-core/apps/authentication-portal/src/main/webapp/fido2-auth.jsp:48]: Fixed with https://github.com/wso2/identity-apps/pull/5833

**Pull request merged**